plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetch.ts

```diff
@@ -76,29 +101,54 @@ export function createFetchCatalogEntityAction(options: {
             title: 'Entity found by the entity reference',
             type: 'object',
             description:
-              'Object containing same values used in the Entity schema.',
+              'Object containing same values used in the Entity schema. Only when used with `entityRef` parameter.',
+          },
+          entities: {
+            title: 'Entities found by the entity references',
+            type: 'array',
+            description:
+              'Array containing objects with same values used in the Entity schema. Only when used with `entityRefs` parameter.',
           },
         },
       },
     },
     async handler(ctx) {
-      const { entityRef, optional } = ctx.input;
-      let entity;
-      try {
-        entity = await catalogClient.getEntityByRef(entityRef, {
+      const { entityRef, entityRefs, optional } = ctx.input;
+      if (!entityRef && !entityRefs) {
+        if (optional) {
+          return;
+        }
+        throw new Error('Missing entity reference or references');
+      }
+
+      if (entityRef) {
+        const entity = await catalogClient.getEntityByRef(entityRef, {
           token: ctx.secrets?.backstageToken,
         });
-      } catch (e) {
-        if (!optional) {
-          throw e;
+
+        if (!entity && !optional) {
+          throw new Error(`Entity ${entityRef} not found`);
         }
+        ctx.output('entity', entity ?? null);
       }
 
-      if (!entity && !optional) {
-        throw new Error(`Entity ${entityRef} not found`);
-      }
+      if (entityRefs) {
+        const entities = await catalogClient.getEntitiesByRefs(
+          { entityRefs },
+          {
+            token: ctx.secrets?.backstageToken,
+          },
+        );
 
-      ctx.output('entity', entity ?? null);
+        const finalEntities = entities.items.map((e, i) => {
+          if (!e && !optional) {
+            throw new Error(`Entity ${entityRefs[i]} not found`);
+          }
+          return e ?? null;
```

Comment:
These are actually nulls when missing already so no need for the fallback `map` at all. You can just make a `for..of` loop that runs when `!optional` to look for missing things, without mapping

Reply:
https://github.com/backstage/backstage/pull/16831